src/archivematicaCommon/tests/test_elasticsearch_functions.py

```diff
@@ -239,3 +239,133 @@ def test_create_indexes_wrong_index(self, mock, patch):
+        "FILEUUID": "44d3aa6d-8bb0-4cfb-bd93-f1121c08916e",
+    },
+    {
+        "filePath": "objects/LEG1363.01.TIF",
+        "FILEUUID": "0552c02b-8626-456f-89cc-bc5f3ce8a112",
+    },
+    {
+        "filePath": "objects/metadata/transfers/MAPS2015-AM641-44f3ee8e-88fd-424c-9d2b-f35d69b148e1/checksum.md5",
+        "FILEUUID": "83aaf419-ec27-4105-b7fd-29df4b07b50e",
+    },
+)
+fileuuid_premisv2_no_ns = (
+    {
+        "filePath": "objects/AM68.csv",
+        "FILEUUID": "fc0e52ca-a688-41c0-a10b-c1d36e21e804",
+    },
+    {
+        "filePath": "objects/V00154.MPG",
+        "FILEUUID": "3a6a182a-40a0-4c2b-9752-fc7e91ac1edf",
+    },
+    {
+        "filePath": "objects/V00158.MPG",
+        "FILEUUID": "431913ba-4379-4373-8798-cc5f2b9dd769",
+    },
+)
+
+
+@pytest.mark.parametrize(
+    "metsfile,fileuuid_dict,filecount,aipuuid,aipname",
+    [
+        (
+            "test_index_fileuuid_METS_premisv3.xml",
+            fileuuid_premisv3,
+            4,
+            "37abc30b-a258-4389-b1f2-67ccd330bc7e",
+            "evelynphotos",
+        ),
+        (
+            "test_index_fileuuid_METS_premisv2.xml",
+            fileuuid_premisv2,
+            4,
+            "9559945a-52e8-4eb4-ac1a-e0e794e758fb",
+            "MAPS2015-AM641",
+        ),
+        (
+            "test_index_fileuuid_METS_premisv2_no_ns.xml",
+            fileuuid_premisv2_no_ns,
+            3,
+            "bdcb560d-7ddd-4c13-8040-1e565b4eddff",
+            "AM68",
+        ),
+    ],
+)
+@mock.patch("elasticSearchFunctions.get_dashboard_uuid")
+@mock.patch("elasticSearchFunctions._wait_for_cluster_yellow_status")
+@mock.patch("elasticSearchFunctions._try_to_index")
+def test_index_aipfile_fileuuid(
+    dummy_try_to_index,
+    dummy_wait_for_cluster_yellow_status,
+    dummy_get_dashboard_uuid,
+    metsfile,
+    fileuuid_dict,
+    filecount,
+    aipuuid,
+    aipname,
+):
+    # Set up mocked up functions
+    # replace _try_to_index() with a mock function that returns the
+    # dict indexed_data, with the fileuuids that _index_aip_files()
+    # obtained from the METS
+
+    dummy_get_dashboard_uuid.return_value = "test-uuid"
+
+    indexed_data = {}
+
+    def get_fileuuid(client, indexData, index, printfn):
+        indexed_data[indexData["filePath"]] = indexData["FILEUUID"]
+
+    dummy_try_to_index.side_effect = get_fileuuid
+
+    mets_file_path = os.path.join(THIS_DIR, "fixtures", metsfile)
+    uuid = aipuuid
+    sipName = "{}-{}".format(aipname, uuid)
+    identifiers = []
+    indexed_files_count = elasticSearchFunctions._index_aip_files(
+        client=None,
+        uuid=uuid,
+        mets_path=mets_file_path,
+        name=sipName,
+        identifiers=identifiers,
+    )
+
+    # check number of indexed files is correct
+    assert indexed_files_count == filecount
+    assert dummy_try_to_index.call_count == filecount
```

Comment:
Are this checks needed for this test? I think we should only check the file UUID parsing to reduce the METS files.